### PR TITLE
feat(gemini): autogenerated update

### PR DESCRIPTION
**Pull request by polsala**
## Rationale
- Automated proposal from the Gemini generator delivering a fresh community utility.

## Why safe to merge
- Utility is isolated to `utils/<util_name>`.
- README + tests ship together (see folder contents).
- No secrets or credentials touched.

## Test Plan
- Follow the instructions in the generated README

## Links
- Generated docs and examples committed alongside this change.

## Mock Justification
- Not applicable; generator did not introduce new mocks.

### CODE REVIEW

**Comment by polsala** [comment]
✅ What’s solid
*   **Clear Purpose and Scope:** The utility is well-defined, addressing a specific, albeit niche, communication need in a post-apocalyptic context. Its self-contained nature is a strength, making it easy to understand and deploy.
*   **Robust Core Logic:** The `caesar_cipher` function correctly handles character casing, wraps around the alphabet using modulo arithmetic, and preserves non-alphabetic characters. This demonstrates a solid implementation of the core encryption/decryption logic.
*   **User-Friendly Interface:** The `argparse` implementation provides a clear and intuitive command-line interface for encoding and decoding messages, complete with helpful descriptions for each argument.
*   **Comprehensive Documentation:** The `README.md` is exceptionally thorough, covering purpose, prerequisites, usage examples, and testing instructions. The examples are particularly helpful for new users.
*   **Automated Generation:** The PR indicates this is an "autogenerated update," which suggests a consistent standard and reduces manual error potential if the generator is reliable.

🧪 Tests
*   **Extensive Unit Tests:** The `test_caesar_cipher` suite covers a wide range of scenarios, including basic encode/decode, wrap-around behavior, handling of non-alphabetic characters, empty strings, zero shifts, and large shifts. This demonstrates excellent coverage for the core logic.
*   **Integration-like Tests for `main`:** The tests for the `main` function effectively simulate command-line execution using `sys.argv` and capture `sys.stdout` and `sys.stderr`, providing confidence in the end-to-end functionality of the script.
*   **Clear Mock Rationale:** The inline comments in `test_whisperer.py` explaining the rationale behind mocking `sys.stdout`, `sys.stderr`, and `sys.argv` are highly valuable for maintainability and understanding.

**Actionable Feedback for Tests:**
*   **Add explicit tests for negative shift values:** While Python's modulo operator handles negative numbers gracefully (e.g., `-3 % 26` results in `23`), explicitly adding test cases for negative shifts would make the test suite even more robust and demonstrate this expected behavior.
    ```python
    # In TestWastelandWhisperer class
    def test_caesar_encode_negative_shift(self):
        self.assertEqual(caesar_cipher("hello", -3, encode=True), "ebiil")
    def test_caesar_decode_negative_shift(self):
        self.assertEqual(caesar_cipher("ebiil", -3, encode=False), "hello")
    ```

🔒 Security
*   **Isolation:** The utility is isolated within its dedicated directory (`utils/nightly-wasteland-whisperer/utils/wasteland-whisperer`), limiting its potential blast radius if vulnerabilities were present.
*   **No External Dependencies/Secrets:** The PR states "No secrets or credentials touched," and the code confirms no external network calls, file I/O (beyond `sys.stdout`/`sys.stderr`), or sensitive operations. This significantly reduces the attack surface.
*   **Transparent Algorithm Limitations:** The `README.md` explicitly states that the Caesar cipher is "not military-grade encryption," which sets appropriate expectations and prevents misuse for highly sensitive data.

**Actionable Feedback for Security:**
*   **Consider input validation for shift key range (optional):** While the modulo operator handles any integer shift value correctly, if there's an intended practical range for the `shift` key (e.g., 0-25), adding a check or a warning for values outside this range could improve user experience, though it's not strictly necessary for cryptographic correctness.

🧩 Docs/DX
*   **Excellent README:** The `README.md` is a standout feature, providing clear, concise, and contextually appropriate documentation. The "Wasteland Whisperer" theme is consistently applied and enhances the user experience.
*   **Clear Usage Examples:** The examples for encoding and decoding are practical and demonstrate the utility's functionality effectively.
*   **Self-Contained:** The utility is easy to run and test, requiring only Python 3.x, which contributes to a positive developer experience.
*   **Informative PR Body:** The PR body provides a good summary of the rationale, safety, and testing approach, which aids in a quick review.

**Actionable Feedback for Docs/DX:**
*   **Clarify `README.md` pathing for execution:** The `README.md` instructs `cd utils/wasteland-whisperer/src` and `cd utils/wasteland-whisperer/tests`. Given the full path `utils/nightly-wasteland-whisperer/utils/wasteland-whisperer/`, these `cd` commands are relative to `utils/nightly-wasteland-whisperer/`. To avoid potential confusion for users starting from the repository root, consider updating the `cd` commands to reflect the full path from the root, or explicitly state the assumed starting directory.
    ```diff
    --- a/utils/nightly-wasteland-whisperer/utils/wasteland-whisperer/README.md
    +++ b/utils/nightly-wasteland-whisperer/utils/wasteland-whisperer/README.md
    @@ -16,14 +16,14 @@
     
     ### Running the Utility
     
-    Navigate to the `src` directory within the `wasteland-whisperer` folder.
+    From the repository root, navigate to the `src` directory:
     
     ```bash
-    cd utils/wasteland-whisperer/src
+    cd utils/nightly-wasteland-whisperer/utils/wasteland-whisperer/src
     ```
     
     #### Encode a Message
    ...
    @@ -58,7 +58,7 @@
     ### Running Tests
     
     Navigate to the `tests` directory and run `unittest`:
-
     ```bash
-    cd utils/wasteland-whisperer/tests
+    cd utils/nightly-wasteland-whisperer/utils/wasteland-whisperer/tests
     python -m unittest test_whisperer.py
     ```
    ```
*   **Consider adding a `pyproject.toml` (optional):** For modern Python projects, including a `pyproject.toml` (even if minimal and only specifying `[project]` metadata) is a best practice. It would explicitly define the project structure and dependencies, even if `argparse` is a standard library module.

🧱 Mocks/Fakes
*   **Appropriate Mocking Strategy:** The use of `unittest.mock.patch` for `sys.stdout`, `sys.stderr`, and `sys.argv` is an excellent and standard approach for testing command-line utilities without actual I/O or modifying the global state.
*   **Clear Rationale:** The inline comments in `test_whisperer.py` explaining *why* specific mocks are used are highly commendable and improve the readability and maintainability of the tests.
*   **Minimal Mocking:** Only necessary components (`sys` streams and arguments) are mocked, avoiding over-mocking and ensuring that the actual logic of `main` is tested.

**Actionable Feedback for Mocks/Fakes:**
*   **No new mocks introduced:** The PR body states "Not applicable; generator did not introduce new mocks." This is accurate as the mocks are for testing the *new* utility, not for existing system components. The existing mocks are well-justified and correctly implemented.
